subprojects/core/src/main/java/org/gradle/api/internal/changedetection/rules/AbstractNamedFileSnapshotTaskStateChanges.java

```diff
@@ -58,35 +53,17 @@ protected AbstractNamedFileSnapshotTaskStateChanges(TaskExecution previous, Task
 
     protected Iterator<TaskStateChange> getFileChanges(final boolean includeAdded) {
         if (getPrevious() == null) {
-            return Iterators.<TaskStateChange>singletonIterator(new DescriptiveChange(title + " file history is not available."));
-        }
-        final List<TaskStateChange> propertyChanges = Lists.newLinkedList();
-        DiffUtil.diff(getCurrent().keySet(), getPrevious().keySet(), new ChangeListener<String>() {
-            @Override
-            public void added(String element) {
-                propertyChanges.add(new DescriptiveChange("%s property '%s' has been added for %s", title, element, task));
-            }
-
-            @Override
-            public void removed(String element) {
-                propertyChanges.add(new DescriptiveChange("%s property '%s' has been removed for %s", title, element, task));
-            }
-
-            @Override
-            public void changed(String element) {
-                // Won't happen
-                throw new AssertionError();
-            }
-        });
-        if (!propertyChanges.isEmpty()) {
-            return propertyChanges.iterator();
+            return Iterators.emptyIterator();
         }
         return Iterators.concat(Iterables.transform(getCurrent().entrySet(), new Function<Map.Entry<String, FileCollectionSnapshot>, Iterator<TaskStateChange>>() {
             @Override
             public Iterator<TaskStateChange> apply(Map.Entry<String, FileCollectionSnapshot> entry) {
                 String propertyName = entry.getKey();
                 FileCollectionSnapshot currentSnapshot = entry.getValue();
                 FileCollectionSnapshot previousSnapshot = getPrevious().get(propertyName);
+                if (previousSnapshot == null) {
+                    return Iterators.emptyIterator();
+                }
```

Comment:
Why is this needed now?

Reply:
The check was there before, too, only earlier (before comparing all the properties).

Reply:
:+1: